src/System.Private.Xml/tests/Xslt/XslTransformApi/CXslTransform.cs

```diff
@@ -2928,6 +2747,13 @@ public void var4(InputType inputType, ReaderType readerType, TransformType trans
         }
     }
 
+    internal sealed class AllowDefaultResolverContext : IDisposable
```

Comment:
Thanks for the link. The issue with the lock in this case is that we need to separate two groups:
1. Tests that change the default app context
2. Tests that assume the default value for it.

In principle tests in each of these groups should be able to run concurrently but concurrent tests from groups 1 and 2 are problematic, so the lock won't protect what we need to be protected.